Microsoft.Toolkit.Uwp.UI.Controls/MosaicControl/MosaicControl.cs

```diff
@@ -0,0 +1,1112 @@
+            }
+
+            if (scrollOrientation == ScrollOrientation.Horizontal || scrollOrientation == ScrollOrientation.Both)
+            {
+                expressionX.Expression = expressionXString;
+
+                if (scrollOrientation == ScrollOrientation.Horizontal)
+                {
+                    // In horizontal mode we never move the offset y
+                    expressionY.Expression = "0";
+                    this.containerVisual.Offset = new Vector3((float)this.OffsetY, 0, 0);
+                }
+            }
+
+            if (scrollOrientation == ScrollOrientation.Vertical || scrollOrientation == ScrollOrientation.Both)
+            {
+                expressionY.Expression = expressionYString;
+
+                if (scrollOrientation == ScrollOrientation.Vertical)
+                {
+                    // In vertical mode we never move the offset x
+                    expressionX.Expression = "0";
+                    this.containerVisual.Offset = new Vector3(0, (float)this.OffsetX, 0);
+                }
+            }
+
+            this.containerVisual.StopAnimation("Offset.X");
+            this.containerVisual.StopAnimation("Offset.Y");
+
+            this.containerVisual.StartAnimation("Offset.X", expressionX);
+            this.containerVisual.StartAnimation("Offset.Y", expressionY);
+
+            this.propertyOffsetModulo = propertyOffsetModulo;
+        }
+
+        private CompositionPropertySet propertyOffsetModulo = null;
+
+        private void RefreshMove()
+        {
+            this.RefreshMove(this.OffsetX + this.animationX, this.OffsetY + this.animationY);
+        }
+
+        object lockerOffset = new object();
+
+        /// <summary>
+        /// Refresh the offset
+        /// </summary>
+        private void RefreshMove(double x, double y)
+        {
+            lock (lockerOffset)
+            {
+                if (this.GetUIStrategy() == UIStrategy.Composition)
+                {
+                    if (propertyOffsetModulo == null)
+                    {
+                        return;
+                    }
+
+                    propertyOffsetModulo.InsertScalar("offsetX", (float)x);
+                    propertyOffsetModulo.InsertScalar("offsetY", (float)y);
+                }
+                else
+                {
+                    var orientation = this.ScrollOrientation;
+
+                    var scrollviewer = this.scrollviewer;
+
+                    double scrollX = 0;
+                    double scrollY = 0;
+
+                    if (scrollviewer != null)
+                    {
+                        var speedRatio = this.ParallaxSpeedRatio;
+
+                        scrollX = -((scrollviewer.HorizontalOffset * scrollviewer.ActualWidth) / scrollviewer.ViewportWidth) * speedRatio;
```

Comment:
Would it better to use the parallax behavior that's going into the Toolkit? (I think there is a PR for it already) This way there is a consistent way for people to adjust the parallax effect across the toolkit. Appreciate that the PR hasn't gone through yet, but thought I would ask the question here in case you weren't aware.